docs/content/preview/migrate/yb-voyager/performance.md

```diff
@@ -10,6 +10,109 @@ menu:
+
+However, we can take choose techniques to reduce the impact on the performance when we are migrating data for the first time into a newly created empty database. For example, yb-voyager assumes that it should be perfectly fine to:
+
+- Create secondary indexes after the data import is complete - YugabyteDB has a feature called [Index Backfill](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/online-index-backfill.md). This feature allows updating indexes after data has been loaded to the main table. This is a lot faster than online index maintenance. The [import data](../../yb-voyager/migrate-steps/#import-data) phase in yb-voyager creates the secondary indexes after it completes data loading on all the tables.
+
+- Disable constraint checks - It is safe to disable all the constraint checks, provided the data is from a reliable source. For maximum throughput, it is also preferred to not follow any order in which tables are populated. yb-voyager therefore, disables all the constraints, except primary key violation errors during the data import phase.
+
+{{< note title="Note" >}}
+
+yb-voyager disables all the constraints only in internal sessions to migrate data.
+
+{{< /note >}}
+
+- Disable Triggers - Triggers are disabled in those sessions where yb-voyager uses to import data in a YugabyteDB cluster, because it results in repeating trigger actions in the new target database which is not required.
+
+## Other Factors
+
+There are other factors which can impact the speed of data ingestion. One or more or all of the following can be done to increase the ingestion performance.
+
+- Data load parallelism - The yb-voyager engine executes ‘N’ parallel batch ingestion jobs at any given time where ‘N’ is equal to the number of nodes in the YugabyteDB cluster. Normally this is considered as a good default practice. However, the target YugabyteDB cluster can be running on high resource machines. Machines which have a large number of cpu cores. In such cases, the default parallelism may seem unfit resulting in underutilization of cpu resources.
+
+It is good practice to have at least ‘C’ number of batch ingestion jobs run simultaneously, where ‘C’ is equal to the sum total of all the cores available in the entire cluster.
+
+{{< note title="Note" >}}
+
+If the cpu usage is going beyond 80%, then it is recommended to lower the parallelism a bit. Similarly if the cpu% is still low, one can increase the parallelism. Include the [–parallel-jobs](../../yb-voyager/yb-voyager-cli/#parallel-jobs) to the import data command to override the default parallelism by specifying the number of connections.
+
+{{< /note >}}
+
+- Batch size - Choosing a very small value of [--batch-size](../../yb-voyager/yb-voyager-cli/#batch-size) (Default is 100000) might result in slow performance, because the time spent on actual data import might be comparable/less to time spent on other tasks such as bookkeeping, setting up the client connection and so on.
```

Comment:
```suggestion
- Batch size - Choosing a very small value for [--batch-size](../../yb-voyager/yb-voyager-cli/#batch-size) (Default is 100000) can result in slow performance, because the time spent on actual data import might be comparable or less than time spent on other tasks such as bookkeeping, setting up the client connection, and so on.
```